src/DynamoCore/Engine/CodeGeneration/AstBuilder.cs

```diff
@@ -331,42 +257,62 @@ private void CompileToAstNodes(NodeModel node, List<AssociativeNode> resultList,
                 }
             }
 
-            if (verboseLogging)
+            if (null == astNodes)
+            {
+                resultList.AddRange(new AssociativeNode[0]);
+            }
+            else if (context == CompilationContext.DeltaExecution)
             {
-                foreach (var n in astNodes)
+                resultList.AddRange(astNodes);
+                if (profilingSession != null)
                 {
-                    Log(n.ToString());
+                    resultList.Add(profilingSession.CreatePostCompilationAstNode(node, inputAstNodes));
                 }
-            }
 
-            if(null == astNodes)
-                resultList.AddRange(new AssociativeNode[0]);
-            else if (context == CompilationContext.DeltaExecution || context == CompilationContext.PreviewGraph)
+                OnAstNodeBuilt(node.GUID, resultList);
+            }
+            else if (context == CompilationContext.PreviewGraph)
             {
-                OnAstNodeBuilt(node.GUID, astNodes);
                 resultList.AddRange(astNodes);
+
+                OnAstNodeBuilt(node.GUID, resultList);
             }
             else if (context == CompilationContext.NodeToCode)
             {
                 resultList.AddRange(astNodes);
             }
-            else //Inside custom node compilation.
+            else
             {
+                // Inside custom node compilation
                 bool notified = false;
                 foreach (var item in astNodes)
                 {
                     if (item is FunctionDefinitionNode)
                     {
                         if (!notified)
+                        {
                             OnAstNodeBuilding(node.GUID);
+                        }
+
                         notified = true;
-                        //Register the function node in global scope with Graph Sync data,
-                        //so that we don't have a function definition inside the function def
-                        //of custom node.
+
+                        // Register the function node in global scope with Graph Sync data,
+                        // so that we don't have a function definition inside the function def
+                        // of custom node.
                         OnAstNodeBuilt(node.GUID, new[] { item });
                     }
                     else
+                    {
                         resultList.Add(item);
+                    }
+                }
+
+                if (verboseLogging)
+                {
+                    foreach (var n in resultList)
+                    {
+                        Log(n.ToString());
```

Comment:
where does this log? - just curious I know this is old code.

Reply:
I'm not really sure, I can find out.